tests/test_settings.py

```diff
@@ -126,208 +128,175 @@ def test_base_path_overrides_config_file_code_and_deps(
-            {},
-            {},
-            ValidationError,
-            id="config_file_unsupported_fields__raises_ValidationError",
-        ),
-        pytest.param(
-            dict(actions="list_imports"),  # actions is not a list
-            {},
-            {},
-            ValidationError,
-            id="config_file_invalid_values__raises_ValidationError",
-        ),
-        pytest.param(
-            dict(actions=["list_deps"], deps=["my_requirements.txt"]),
-            {},
-            {},
-            make_settings_dict(
-                actions={Action.LIST_DEPS}, deps={Path("my_requirements.txt")}
-            ),
-            id="config_file__overrides_some_defaults",
-        ),
-        pytest.param(
-            None,
-            dict(actions="list_imports"),  # actions is not a list
-            {},
-            SettingsError,
-            id="env_var_with_wrong_type__raises_SettingsError",
-        ),
-        pytest.param(
-            None,
-            dict(ignore_unused='["foo", "missing_quote]'),  # cannot parse value
-            {},
-            SettingsError,
-            id="env_var_with_invalid_value__raises_SettingsError",
-        ),
-        pytest.param(
-            None,
-            dict(actions='["list_imports"]', ignore_unused='["foo", "bar"]'),
-            {},
-            make_settings_dict(
-                actions={Action.LIST_IMPORTS}, ignore_unused={"foo", "bar"}
-            ),
-            id="env_vars__overrides_some_defaults",
-        ),
-        pytest.param(
-            dict(code=["my_code_dir"], deps=["my_requirements.txt"]),
-            dict(actions='["list_imports"]', ignore_unused='["foo", "bar"]'),
-            {},
-            make_settings_dict(
-                actions={Action.LIST_IMPORTS},
-                code={Path("my_code_dir")},
-                deps={Path("my_requirements.txt")},
-                ignore_unused={"foo", "bar"},
-            ),
-            id="config_file_and_env_vars__overrides_separate_defaults",
-        ),
-        pytest.param(
-            dict(code="my_code_dir", deps=["my_requirements.txt"]),
-            dict(actions='["list_imports"]', code='["<stdin>"]'),
-            {},
-            make_settings_dict(
-                actions={Action.LIST_IMPORTS},
-                code={"<stdin>"},
-                deps={Path("my_requirements.txt")},
-            ),
-            id="config_file_and_env_vars__env_overrides_file",
+@dataclass
+class SettingsTestVector:
+    """Test vectors for FawltyDeps Settings configuration."""
+
+    id: str
+    config_settings: Optional[Union[str, TomlData]] = None
+    env_settings: Dict[str, Any] = field(default_factory=dict)
+    cmdline_settings: Dict[str, Any] = field(default_factory=dict)
+    expect: Union[Dict[str, Any], Type[Exception]] = field(
+        default_factory=lambda: EXPECT_DEFAULTS
+    )
+
+
+settings_test_vector = [
```

Comment:
Nit:

It feels odd to have a vector of `SettingsTestVector` itself be called `settings_test_vector`.